### DISCUSSION TITLE
Issue template for organization does not work

### DISCUSSION THREAD

**Discussion by nikolaas** [community/community#22451] [New to GitHub]
<p>Hello.</p>
<p>I try to create issue templates for my organization’s projects in accordance with the article “<a href="https://help.github.com/en/articles/creating-a-default-community-health-file-for-your-organization" rel="nofollow noopener">Creating a default community health file for your organization</a>”.</p>
<ol>
<li>
<p>I created a public repository called “.github”.</p>
</li>
<li>
<p>Then I created a directory called “ISSUE_TEMPLATE” in the root of the repository.</p>
</li>
<li>
<p>Then I created files “bug_report.md” and “feature_request.md” and placed them in the directory “ISSUE_TEMPLATE”.</p>
</li>
<li>
<p>After that, I tried to create an issue with the button “New issue” in another organization’s repository.</p>
</li>
</ol>
<p>I expected to see the next page:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/587ef367df7a6bbdffa782ffb80073ebe4f8afe2.png" data-download-href="/uploads/short-url/cCS0vGjTGybTsrXD7LnS2fwKfuO.png?dl=1" title="new-issue-page-with-multiple-templates.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6443153/181071011-73988837-2bdf-4643-8b38-e77713fb8d14.png" alt="new-issue-page-with-multiple-templates.png" data-base62-sha1="cCS0vGjTGybTsrXD7LnS2fwKfuO" width="690" height="298" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/587ef367df7a6bbdffa782ffb80073ebe4f8afe2_2_690x298.png, https://aws1.discourse-cdn.com/github/optimized/1X/587ef367df7a6bbdffa782ffb80073ebe4f8afe2_2_1035x447.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/587ef367df7a6bbdffa782ffb80073ebe4f8afe2_2_1380x596.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/587ef367df7a6bbdffa782ffb80073ebe4f8afe2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">new-issue-page-with-multiple-templates.png</span><span class="informations">1604×694 145 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Instead, I see the next page:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/63189b5055f2d293a86c7b595b968f1894e499a9.png" data-download-href="/uploads/short-url/e8DZknXXRLah2yYAVBivNWpsNfj.png?dl=1" title="gitgub_new_issue.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/6443153/181071013-eed33863-71fb-4cbd-bd0d-2770156467d3.png" alt="gitgub_new_issue.png" data-base62-sha1="e8DZknXXRLah2yYAVBivNWpsNfj" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/63189b5055f2d293a86c7b595b968f1894e499a9_2_690x362.png, https://aws1.discourse-cdn.com/github/original/1X/63189b5055f2d293a86c7b595b968f1894e499a9.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/63189b5055f2d293a86c7b595b968f1894e499a9.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/63189b5055f2d293a86c7b595b968f1894e499a9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">gitgub_new_issue.png</span><span class="informations">1021×536 46.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Tell me please, did I do something wrong? Or maybe is it a bug?</p>

**Comment by mpboom** [comment]
<p>Hi <a class="mention" href="/nikolaas">@nikolaas</a> ,</p>
<p>No, you didn’t do anything wrong. Extending default issue/PR templates across an account or organization is not (yet) supported according to the document you linked yourself.</p>

**Comment by nikolaas** [comment]
<p><a class="mention" href="/mpboom">@mpboom</a>, thank you for the reply.</p>
<p>But I am confused. The article says in the start:</p>
<blockquote>
<p>You can create default community health files, such as CONTRIBUTING and CODE_OF_CONDUCT, for your organization. Default files will be used for any public repository in your organization that does not contain its own file of that type.</p>
</blockquote>
<p>Then it says in the section “<a href="https://help.github.com/en/articles/creating-a-default-community-health-file-for-your-organization#supported-file-types" rel="nofollow noopener">Supported file types</a>”:</p>
<blockquote>
<p>You can create organization-wide defaults for the following community health files:</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<table><tbody><tr>
<td>Issue and pull request templates</td>
<td>Issue and pull request templates customize and standardize the information you'd like contributors to include when they open issues and pull requests in your repository. For more information, see "About issue and pull request templates."</td>
</tr></tbody></table>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>You cannot create a default license file for your organization.</p>
</blockquote>
<p>My understanding, GitHub does not support the file “LICENSE” as default community health files for an organization. But files with the rest types including issue/PR templates are supported.</p>
<p>Why do you say that extending default issue/PR templates across an organization is not supported according to the article?</p>

**Comment by mpboom** [comment]
<p>Ouch, I seem to have misread the article a bit… I’m sorry - you indeed seem to be right.</p>

**Comment by mpboom** [comment]
<p>However in your original post you seem to be opening an issue in a project that isn’t yours. These things only work for projects within your own organization. So I suppose that is a reason why it still doesn’t work.</p>

**Comment by nikolaas** [comment]
<p>This is a repository with community health files: <a href="https://github.com/i-core/.github" rel="nofollow noopener">https://github.com/i-core/.github</a>.</p>
<p>This is a repository in which I want to create an issue: <a href="https://github.com/i-core/reqlog" rel="nofollow noopener">https://github.com/i-core/reqlog</a>.</p>
<p>Both repositories belong to the organization <a href="https://github.com/i-core" rel="nofollow noopener">https://github.com/i-core</a>. But I can’t use the default issue template that is placed in the repository “.github” when I create an issue in the repository “reqlog”.</p>

**Comment by aiwebb** [comment, marked as answer]
<p>We also experienced this issue.</p>
<p>We resolved it by placing the ISSUE_TEMPLATE directory underneath a <strong>.github</strong>  directory of the <strong>.github</strong>  repository in the organization, e.g <strong>myorg/.github/.github/ISSUE_TEMPLATE/bug_report.md</strong>. This does cause other repos in the same org to use those configured templates.</p>
<p>It’s an extra level of <strong>.github</strong> versus what the documentation says (emphasis added):</p>
<blockquote>
<p>You can add default community health files for your organization <strong>to the root</strong> of a public repository called .github that is owned by the organization.</p>
</blockquote>

**Reply by ritrafa** [reply]
Make sure that your .github repo is public for this to work (others noted below as well)

**Reply by sbarbett** [reply]
I spent hours trying to figure this out this morning and this was the only thing that worked. If nesting the ISSUE_TEMPLATE folder inside a .github directory within the .github org repository is required, I wish the documentation were clearer about this.

**Comment by mpboom** [comment]
<p>Thanks a lot for taking the effort to answer the question <a class="mention" href="/aiwebb">@aiwebb</a> ! <a class="mention" href="/nikolaas">@nikolaas</a> , does that help out? If so, give <a class="mention" href="/aiwebb">@aiwebb</a> some kudos and mark his solution as the accepted solution!</p>

**Comment by nikolaas** [comment]
<p><a class="mention" href="/aiwebb">@aiwebb</a>, thanks for your help.</p>
<p>I moved the directory <strong>ISSUE_TEMPLATE</strong>  underneath the  <strong>.github</strong>  directory of the  <strong>.github</strong>  repository. After that, the issue templates started working.</p>
<p>I tried to move the directory <strong>PULL_REQUEST_TEMPLATE</strong> to the directory <strong>.github</strong> but it does not help out. Since my directory <strong>PULL_REQUEST_TEMPLATE</strong> contains only a file <strong>pull_request_template.md</strong> I renamed the file to <strong>PULL_REQUEST_TEMPLATE.md</strong> and moved it directly to the directory <strong>.github</strong>. After that, the default pull request template started working.</p>
<p><a class="mention" href="/mpboom">@mpboom</a>, thank you as well. I think it would be nice to align the documentation and the implementation of this feature.</p>

**Comment by mpboom** [comment]
<p>Great it worked! I agree that the documentation should be aligned, but am not affiliated with GitHub myself. I’ve alerted one of the Community Moderators; they’ll probably be in touch soon with the documentation team.</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi everyone!</p>
<p>Thanks for letting us know about this confusing part of the documentation. I’ve passed on your feedback to our documentation team.</p>
<p>Thanks!</p>

**Comment by neubs-bsi** [comment]
<p>Hello <a class="mention" href="/that-pat">@that-pat</a> ,</p>
<p>I wanted to inquire what the state of this is, since in the <a href="https://help.github.com/en/articles/creating-a-pull-request-template-for-your-repository" rel="nofollow noopener">documentation</a> it is still stated that the path</p>
<pre><code>.github/PULL_REQUEST_TEMPLATE/some_pull_request_template.md
</code></pre>
<p>should be valid. However, it is not working for our <a href="https://github.com/eclipse/antenna" rel="nofollow noopener">project</a> and in several forums I have found online as well.</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/neubs-bsi">@neubs-bsi</a>,</p>
<p>I see that you have a Pull Request template set up on this project that is working, but it’s using the single pull request template format instead of the one for multiple formats. Do you have an example where this is still not working so that I can reproduce the issue in its current state?</p>
<p>Thanks! </p>

**Comment by neubs-bsi** [comment]
<p>Hello <a class="mention" href="/that-pat">@that-pat</a> ,</p>
<p>thanks for taking my post into consideration.</p>
<p>I have created a dummy project, that attempts to use two pull request templates:</p>
<p><a href="https://github.com/neubs-bsi/pullRequestTemplateDummyProject" rel="nofollow noopener">https://github.com/neubs-bsi/pullRequestTemplateDummyProject</a></p>
<p>The file structure is as promoted in the documentation and I have created a pull request testing the feature:</p>
<p><a href="https://github.com/neubs-bsi/pullRequestTemplateDummyProject/pull/1" rel="nofollow noopener">https://github.com/neubs-bsi/pullRequestTemplateDummyProject/pull/1</a></p>
<p>No template appeared.</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/neubs-bsi">@neubs-bsi</a>,</p>
<p>How did you create the pull request? Unlike with issue templates, there is a special link you would use to activate a pull request template.</p>
<p><a href="https://github.com/" rel="nofollow noopener">https://github.com/</a> <strong>&lt;owner&gt;</strong> / <strong>&lt;repository&gt;</strong> /compare/master… <strong>&lt;branch-name&gt;</strong>?quick_pull=1&amp;template= <strong>&lt;template_file&gt;</strong></p>
<p>Thanks!</p>

**Comment by neubs-bsi** [comment]
<p>Hello <a class="mention" href="/that-pat">@that-pat</a> ,</p>
<p>you are right. That did in fact work. Thanks for the help.</p>
<p>For other people having this issue it might be worth putting this explicitly into the documentation of the pull request templates, to make it easier to understand.</p>
<p>Additionally, if there would be an option for choosing a template when creating a pull request via the interface button in github, I would find that very helpful. This could be implemented similarly to the way the choice of the issue templates is implemented.</p>
<p>Kind regards :slight_smile:</p>

**Comment by FantasticFiasco** [comment]
<p>Hi <a class="mention" href="/that-pat">@that-pat</a>!</p>
<p>I seem to have the same problem of not being able to configure the default issue templates. To me my repositories look identical to other user’s where the feature is working. Can you by taking a quick glanze see what I am missing? I feel so stupid.</p>
<p>Default repository: <a href="https://github.com/FantasticFiasco/.github" rel="nofollow noopener">https://github.com/FantasticFiasco/.github</a></p>
<p>Respoitory where issue templates aren’t working: <a href="https://github.com/FantasticFiasco/temp" rel="nofollow noopener">https://github.com/FantasticFiasco/temp</a></p>
<p>Thanks!</p>

**Comment by FantasticFiasco** [comment]
<p>Oh, perhaps it hasn’t been implemented: <a href="https://github.community/t/default-community-files-ignores-issue-templates/2802/4">https://github.community/t5/How-to-use-Git-and-GitHub/Default-community-files-ignores-issue-templates/m-p/43809</a></p>

**Comment by has12zen** [comment]
<p>My issue was similar. I think it was occurring because my_repo inside the organization was private. Once it was made public it started working.</p>

**Reply by matthewford** [reply]
Thanks, this was our issue.
